pkg/client/unversioned/record/event_test.go

```diff
@@ -286,24 +289,24 @@ func TestEventf(t *testing.T) {
 				if !item.expectUpdate {
 					t.Errorf("Expected event create(), got event update()")
 				}
-				called <- struct{}{}
+				wg.Done()
 				return returnEvent, nil
 			},
 		}
 		eventBroadcaster := NewBroadcaster()
 		sinkWatcher := eventBroadcaster.StartRecordingToSink(&testEvents)
 		logWatcher1 := eventBroadcaster.StartLogging(t.Logf) // Prove that it is useful
+		wg.Add(1)
```

Comment:
A waitgroup is simpler to use and better (e.g. it lets goroutines die sooner).

The original code did the waiting right before the Stop() calls, so I did the same with a waitgroup. Splitting the waiting in two separate blocks of code would confuse things, especially since you cannot know which of the two channel receives would correspond to which of the channel sends.